src/kubernetes.cc

```diff
@@ -1078,6 +1078,7 @@ void KubernetesReader::PodCallback(
 void KubernetesReader::WatchPods(MetadataUpdater::UpdateCallback callback)
     const {
   LOG(INFO) << "Watch thread (pods) started";
+  HealthReporter healthReporter;
```

Comment:
Should there not be a single shared health reporter throughout the whole agent binary?